metaflow/plugins/aws/awslambda/lambda_cli.py

```diff
@@ -0,0 +1,204 @@
+)
+@click.pass_context
+def step(
+    ctx,
+    step_name,
+    code_package_sha,
+    code_package_url,
+    run_time_limit,
+    force_fetch_image,
+    memory=None,
+    image=None,
+    **kwargs
+):
+    # type: (...) -> None
+
+    def echo(msg, err):  # type (str, bool) -> None
+        ctx.obj.echo_always("%s" % (msg,), err=err)
+
+    if ctx.obj.datastore.datastore_root is None:
+        ctx.obj.datastore.datastore_root = (
+            ctx.obj.datastore.get_datastore_root_from_config(echo)
+        )
+
+    if R.use_r():
+        entrypoint = R.entrypoint()
+    else:
+        executable = ctx.obj.environment.executable(step_name)
+        entrypoint = "%s -u %s" % (executable, os.path.basename(sys.argv[0]))
+
+    top_args = " ".join(util.dict_to_cli_options(ctx.parent.parent.params))
+
+    input_paths = kwargs.get("input_paths")
+    split_vars = None
+    if input_paths:
+        max_size = 30 * 1024
+        split_vars = {
+            "METAFLOW_INPUT_PATHS_%d" % (i // max_size): input_paths[i : i + max_size]
+            for i in range(0, len(input_paths), max_size)
+        }
+        kwargs["input_paths"] = "".join("${%s}" % s for s in split_vars.keys())
+
+    step_args = " ".join(util.dict_to_cli_options(kwargs))
+    step_cli = u"{entrypoint} {top_args} step {step} {step_args}".format(
+        entrypoint=entrypoint, top_args=top_args, step=step_name, step_args=step_args
+    )
+    node = ctx.obj.graph[step_name]
+
+    # Get retry information. TODO: actually use it somewhere.
+    retry_count = kwargs.get("retry_count", 0)
+
+    task_spec = TaskSpec(
+        ctx.obj.flow.name, step_name, kwargs["run_id"], kwargs["task_id"], retry_count
+    )
+
+    # Set attributes
+    attrs = {}  # type: Dict[str, Optional[str]]
+    attrs.update(task_spec.to_dict(key_prefix="metaflow."))
+    attrs["metaflow.user"] = util.get_username()
+    attrs["metaflow.version"] = ctx.obj.environment.get_environment_info()[
+        "metaflow_version"
+    ]
+
+    env_deco = [deco for deco in node.decorators if deco.name == "environment"]
+    if env_deco:
+        env = env_deco[0].attributes["vars"]
+    else:
+        env = {}
+
+    datastore_root = task_spec.get_datastore_root(ctx.obj.datastore)
+
+    # Add the environment variables related to the input-paths argument
+    if split_vars:
+        env.update(split_vars)
+
+    config = Config(
+        retries={
+            "max_attempts": LAMBDA_THROTTLE_RETRIES,
+        }
+    )
+    lambda_client = boto3.client("lambda", config=config)
+    lambda_arn = ensure_lambda(
+        lambda_client,
+        flow_name=ctx.obj.flow.name,
+        step_name=step_name,
+        memory_mb=memory,
+        timeout_seconds=run_time_limit,
+        role_arn=LAMBDA_ROLE_ARN,
+        image_uri=image or LAMBDA_IMAGE_URI,
+        force_fetch_image=force_fetch_image,
+        echo=echo,
+    )
+    runner = LambdaRunner(
```

Comment:
Given that the `lambda_client` is only used by the `LambdaRunner` - would it make sense to instantiate it within the `LambdaRunner` itself?